definitions/output/all/backfill_requests.js

```diff
@@ -0,0 +1,194 @@
+  }
+""";
+
+CREATE TEMP FUNCTION get_type(mime_typ STRING, ext STRING)
+RETURNS STRING
+LANGUAGE js
+AS """
+  try {
+    mime_typ = mime_typ.toLowerCase();
+
+    // Order by most unique types first
+    const uniqueTypes = ["font", "css", "image", "script", "video", "audio", "xml"];
+    for (let typ of uniqueTypes) {
+      if (mime_typ.includes(typ)) {
+        return typ;
+      }
+    }
+
+    // Special cases
+    if (mime_typ.includes("json") || ["js", "json"].includes(ext)) {
+      return "script";
+    } else if (["eot", "ttf", "woff", "woff2", "otf"].includes(ext)) {
+      return "font";
+    } else if (
+      ["png", "gif", "jpg", "jpeg", "webp", "ico", "svg", "avif", "jxl", "heic", "heif"].includes(ext)
+    ) {
+      return "image";
+    } else if (ext === "css") {
+      return "css";
+    } else if (ext === "xml") {
+      return "xml";
+    } else if (
+      ["mp4", "webm", "ts", "m4v", "m4s", "mov", "ogv", "swf", "f4v", "flv"].includes(ext) ||
+      ["flash", "webm", "mp4", "flv"].some(typ => mime_typ.includes(typ))
+    ) {
+      return "video";
+    } else if (mime_typ.includes("wasm") || ext === "wasm") {
+      return "wasm";
+    } else if (mime_typ.includes("html") || ["html", "htm"].includes(ext)) {
+      return "html";
+    } else if (mime_typ.includes("text")) {
+      // Put "text" last because it is often misused, so extension should take precedence.
+      return "text";
+    } else {
+      return "other";
+    }
+  } catch (e) {
+    return "other"; // Return "other" if there's any error
+  }
+""";
+
+CREATE TEMP FUNCTION parse_headers(headers STRING)
+RETURNS ARRAY<STRUCT<name STRING, value STRING>>
+LANGUAGE js
+AS """
+  try {
+    return JSON.parse(headers).map(header => {
+      return { name: header.name, value: header.value };
+    });
+  } catch (e) {
+    return [];
+  }
+""";
+
+INSERT INTO \`all_dev.requests_stable\` --${ctx.resolve('all', 'requests')}
+SELECT
+  DATE('${iteration.date}') AS date,
+  '${iteration.client}' AS client,
+  requests.page AS page,
+  TRUE AS is_root_page,
+  requests.page AS root_page,
+  crux.rank AS rank,
+  requests.url AS url,
+  IF(
+    SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._request_type') AS STRING) = "Document" AND
+      MIN(SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._index') AS INT64)) OVER (PARTITION BY requests.page) = SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._index') AS INT64),
+    TRUE,
+    FALSE
+  ) AS is_main_document,
+  get_type(JSON_VALUE(requests.payload, '$.response.content.mimeType'), get_ext_from_url(requests.url)) AS type,
+  SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._index') AS INT64) AS index,
+  SAFE.PARSE_JSON(requests.payload, wide_number_mode => 'round') AS payload,
+  TO_JSON( STRUCT(
+    SAFE_CAST(JSON_VALUE(requests.payload, '$.time') AS INTEGER) AS time,
+    JSON_VALUE(requests.payload, '$._method') AS method,
+    NULL AS redirectUrl,
+    JSON_VALUE(requests.payload, '$.request.httpVersion') AS reqHttpVersion,
```

Comment:
@tunetheweb Here is the source for the HTTP version you've mentioned in https://github.com/HTTPArchive/bigquery/issues/110.
Please update it (and `respHttpVersion`) to whatever is the best one.

Note: this query is expected to populate the period from Jan 2016 till Mar 2022.

Reply:
`$._protocol` works better, where it's available (which is why the HTTP/2 query uses that).

```suggestion
    IFNULL(JSON_VALUE(requests.payload, '$._protocol'), JSON_VALUE(requests.payload, '$.request.httpVersion')) AS reqHttpVersion,
```

Compare these two queries:

```sql
SELECT
  JSON_VALUE(payload, '$.request.httpVersion') AS http_version,
  COUNT(0) AS total
FROM
  `httparchive.requests.2020_09_01_desktop`
GROUP BY
  http_version
ORDER BY
  total DESC;

SELECT
  JSON_VALUE(payload, '$._protocol') AS http_version,
  COUNT(0) AS total
FROM
  `httparchive.requests.2020_09_01_desktop`
GROUP BY
  http_version
ORDER BY
  total DESC
```

We also need the combo of both as for older datasets `$._protocol` doesn't exist (e.g. if you try above queries for `2016_01_01` only the first one returns results.